packages/notebook/test/actions.spec.ts

```diff
@@ -1073,27 +1074,39 @@ describe('@jupyterlab/notebook', () => {
     describe('#selectAbove()', () => {
       it('should select the cell above the active cell', async () => {
         widget.activeCellIndex = 1;
-        await NotebookActions.selectAbove(widget);
+        const waitForChange = signalToPromise(widget.stateChanged);
+        NotebookActions.selectAbove(widget);
+        await waitForChange;
         expect(widget.activeCellIndex).toBe(0);
       });
 
-      it('should be a no-op if there is no model', async () => {
+      it.failing('should be a no-op if there is no model', async () => {
         widget.model = null;
-        await NotebookActions.selectAbove(widget);
-        expect(widget.activeCellIndex).toBe(-1);
+        const waitForChange = signalToPromise(widget.stateChanged);
+        NotebookActions.selectAbove(widget);
+        // should timeout as there is no change.
+        await waitForChange;
+        expect(widget.activeCellIndex).toBe(0);
       });
 
-      it('should not wrap around to the bottom', async () => {
-        await NotebookActions.selectAbove(widget);
+      it.failing('should not wrap around to the bottom', async () => {
```

Comment:
Some tests are expected to fail because of timeout while awaiting the `stateChange` signal.
This was the only way I found to test that no changes occur without using async function, but the failing does not take into account the error thrown.

I'm interested in any better solution.